crates/sui-framework/packages/move-stdlib/sources/string.move

```diff
@@ -44,11 +46,20 @@ module std::string {
         }
     }
 
+    /// Alias to align the API with `ASCII`.
+    public use fun bytes as String.as_bytes;
```

Comment:
Do we want an alias here or a new function? 

I suppose the alternative I'm suggesting is 
```
    public fun as_bytes(s: &String): &vector<u8> {
       &s.bytes
    }
    
    public fun bytes(s: &String): &vector<u8> {
        s.as_bytes()
    }
```
I feel like this will better for marking `bytes` as deprecated/hidden, since if you are not on 2024 edition, you can still use the correct function/name

Reply:
K, I added new functions and commented old ones as deprecated.